### PR TITLE
Add DPDK OVS functional test

### PR DESCRIPTION
**Pull request by ader1990**
## Description

Added DPDK enabled openvswitch functional test: VERIFY-DPDK-OVS, to verify DPDK + Openvswitch works on Azure.

This test case builds DPDK, builds and runs OVS with DPDK support and creates an OVS bridge where it adds a DPDK port based on the Mellanox VF.

---

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes in the description.
- [ ] Verify PR checker results and fix any coding errors.
- [ ] Included LISAv2 sample test results to demonstrate code functionality.

### Quality of Code and Contribution Guidelines
- [ ] I have read the [contribution guidelines](https://github.com/LIS/LISAv2/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: Testscripts/Linux/dpdk_ovs_setup.sh [hunk 1]
```diff
@@ -0,0 +1,114 @@
+#!/bin/bash
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache License.
+
+# This script will do setup huge pages
```
Comment (juhlee-microsoft) [inline Testscripts/Linux/dpdk_ovs_setup.sh]: those 2 bash scripts are pretty similar structure. Any chance to integrate each other?

File: Testscripts/Linux/ovs_setup.sh [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+}
+
+function install_ovs () {
+	SetTestStateRunning
+	LogMsg "Configuring ${1} ${DISTRO_NAME} ${DISTRO_VERSION} for OVS test..."
+	packages=(gcc make git tar wget dos2unix psmisc make)
+	case "${DISTRO_NAME}" in
+		ubuntu|debian)
+			ssh "${1}" "until dpkg --force-all --configure -a; sleep 10; do echo 'Trying again...'; done"
+			if [[ "${DISTRO_VERSION}" != "18.04" ]];
```
[the remark this answers is not in the document]
Comment (ader1990) [reply Testscripts/Linux/ovs_setup.sh]: The OS+ OVS + DPDK + KERNEL matrix currently supports only Ubuntu 18.04 + OVS master + DPDK master and linux kernel >= 4.18.

File: Testscripts/Linux/ovs_setup.sh [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+			then
+				echo "Distro unsupported ${DISTRO_VERSION}"
+				SetTestStateAborted
+				exit 1
+			fi
+			ssh "${1}" ". ${UTIL_FILE} && update_repos"
+			packages+=(autoconf libtool)
+			;;
+		*)
```
[the remark this answers is not in the document]
Comment (ader1990) [reply Testscripts/Linux/ovs_setup.sh]: see above

File: Testscripts/Windows/VERIFY-DPDK-OVS.ps1 [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+		Write-LogInfo "SERVER $DataPath NIC: $serverNicName"
+
+		Write-LogInfo "Generating constants.sh ..."
+		$constantsFile = "$LogDir\constants.sh"
+		Set-Content -Value "#Generated by Azure Automation." -Path $constantsFile
+		Add-Content -Value "vms=$($serverVMData.RoleName),$($clientVMData.RoleName)" -Path $constantsFile
+		Add-Content -Value "server=$($serverVMData.InternalIP)" -Path $constantsFile
+		Add-Content -Value "client=$($clientVMData.InternalIP)" -Path $constantsFile
+		Add-Content -Value "nicName=eth1" -Path $constantsFile
+		Add-Content -Value "pciAddress=0002:00:02.0" -Path $constantsFile
```
[the remark this answers is not in the document]
Comment (ader1990) [reply Testscripts/Windows/VERIFY-DPDK-OVS.ps1]: I will remove this line, is not used.

File: Testscripts/Windows/VERIFY-DPDK-OVS.ps1 [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+			}
+		}
+		$detectedDistro = Detect-LinuxDistro -VIP $vmData.PublicIP -SSHport $vmData.SSHPort `
+				-testVMUser $user -testVMPassword $password
+		$currentKernelVersion = Run-LinuxCmd -ip $vmData.PublicIP -port $vmData.SSHPort `
+				-username $user -password $password -command "uname -r"
+		if (IsGreaterKernelVersion -actualKernelVersion $currentKernelVersion -detectedDistro $detectedDistro) {
+				Write-LogInfo "Confirmed Kernel version supported: $currentKernelVersion"
+		} else {
+			Write-LogErr "Unsupported Kernel version: $currentKernelVersion"
```
[the remark this answers is not in the document]
Comment (ader1990) [reply Testscripts/Windows/VERIFY-DPDK-OVS.ps1]: throw does not log

File: Testscripts/Linux/dpdk_ovs_setup.sh (hunk 1 above)
Comment (ader1990) [reply Testscripts/Linux/dpdk_ovs_setup.sh]: they do different things, once installs dpdk in a generic way and the second installs ovs

File: Testscripts/Windows/VERIFY-DPDK-OVS.ps1 [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+			if ($param -imatch "modes") {
+				$modes = ($param.Replace("modes=",""))
+			}
+		}
+		$detectedDistro = Detect-LinuxDistro -VIP $vmData.PublicIP -SSHport $vmData.SSHPort `
+				-testVMUser $user -testVMPassword $password
+		$currentKernelVersion = Run-LinuxCmd -ip $vmData.PublicIP -port $vmData.SSHPort `
+				-username $user -password $password -command "uname -r"
+		if (IsGreaterKernelVersion -actualKernelVersion $currentKernelVersion -detectedDistro $detectedDistro) {
+				Write-LogInfo "Confirmed Kernel version supported: $currentKernelVersion"
```
[the remark this answers is not in the document]
Comment (ader1990) [reply Testscripts/Windows/VERIFY-DPDK-OVS.ps1]: done

File: Testscripts/Windows/VERIFY-DPDK-OVS.ps1 (hunk 5 above)
[the remark this answers is not in the document]
Comment (ader1990) [reply Testscripts/Windows/VERIFY-DPDK-OVS.ps1]: refactored the message

**Pull request merged**